File: examples/test-projects/small-projects/test-project-python/counter.py
"""Счётчик с порогом — аналог Counter из других малых проектов."""

THRESHOLD = 10


class Counter:
    """Счётчик, уведомляющий при изменении значения и достижении порога."""

    def __init__(self, initial=0):
        self.value = initial

    def value_changed(self, v):
        pass

    def threshold_reached(self, t):
        pass

    def get_value(self):
        return self.value

    def set_value(self, new_value):
        if self.value == new_value:
            return
        self.value = new_value
        self.value_changed(self.value)
        if self.value >= THRESHOLD:
            self.threshold_reached(THRESHOLD)

    def increment(self):
        self.set_value(self.value + 1)

    def reset(self):
        self.set_value(0)

File: examples/test-projects/small-projects/test-project-php/Counter.php
<?php
/** Счётчик с порогом — аналог Counter из других малых проектов. */

define('THRESHOLD', 10);

class Counter {
    private int $value;

    public function __construct(int $initial = 0) {
        $this->value = $initial;
    }

    public function valueChanged(int $v): void {
        // hook: вызывается при изменении значения
    }

    public function thresholdReached(int $t): void {
        // hook: вызывается при достижении порога
    }

    public function getValue(): int {
        return $this->value;
    }

    public function setValue(int $newValue): void {
        if ($this->value === $newValue) {
            return;
        }
        $this->value = $newValue;
        $this->valueChanged($this->value);
        if ($this->value >= THRESHOLD) {
            $this->thresholdReached(THRESHOLD);
        }
    }

    public function increment(): void {
        $this->setValue($this->value + 1);
    }

    public function reset(): void {
        $this->setValue(0);
    }
}
